pkg/gadgets/trace/exec/tracer/tracer_test.go

```diff
@@ -0,0 +1,214 @@
+//go:build linux
+// +build linux
+
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer_test
+
+import (
+	"os/exec"
+	"testing"
+
+	utilstest "github.com/kinvolk/inspektor-gadget/pkg/gadgets/internal/test"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/tracer"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/exec/types"
+	eventtypes "github.com/kinvolk/inspektor-gadget/pkg/types"
+)
+
+func TestCreateTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	eventCallback := func(types.Event) {}
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil, eventCallback)
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(func() { tracer.Stop() })
+}
+
+func TestStopTracerIdempotent(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	eventCallback := func(types.Event) {}
+
+	tracer, err := tracer.NewTracer(&tracer.Config{}, nil, eventCallback)
+	if err != nil {
+		t.Fatalf("Error creating tracer: %s", err)
+	}
+	if tracer == nil {
+		t.Fatal("Returned tracer was nil")
+	}
+	t.Cleanup(func() { tracer.Stop() })
+
+	// Check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func TestTracer(t *testing.T) {
+	utilstest.RequireRoot(t)
+
+	// Variable to save the value of the created process PID
+	var catPid int
+
+	manyArgs := []string{}
+	// 19 is DEFAULT_MAXARGS - 1 (-1 because args[0] is on the first position).
+	for i := 0; i < 19; i++ {
+		manyArgs = append(manyArgs, "/dev/null")
+	}
+
+	// Function to generate an event used most of the times
+	generateEvent := func() error {
+		cmd := exec.Command("/bin/cat", "/dev/null")
+		if err := cmd.Run(); err != nil {
+			return err
+		}
+
+		catPid = cmd.Process.Pid
+
+		return nil
+	}
+
+	testRunner := utilstest.NewTestRunner[types.Event](t)
+
+	tests := map[string]utilstest.Test[types.Event]{
```

Comment:
Even if I agree, it seems to much at this point. These tests intend to test the "C + Go" code as a whole. I call them unit tests because in this case the "unit" refers to the gadget. We call integration tests to the ones that test the whole thing flow from getting the event in a kubernetes node to displaying it to the user.